Content.Client/Revolutionary/RevolutionarySystem.cs

```diff
@@ -1,35 +1,60 @@
 using Content.Shared.Revolutionary.Components;
 using Content.Client.Antag;
+using Content.Shared.CCVar;
+using Content.Shared.Ghost;
+using Content.Shared.StatusIcon;
 using Content.Shared.StatusIcon.Components;
+using Robust.Shared.Configuration;
 
 namespace Content.Client.Revolutionary;
 
 /// <summary>
 /// Used for the client to get status icons from other revs.
 /// </summary>
-public sealed class RevolutionarySystem : AntagStatusIconSystem<RevolutionaryComponent>
+public sealed class RevolutionarySystem : SharedStatusIconSystem
 {
+    [Dependency] private readonly IConfigurationManager _cfg = default!;
+
+    private bool _revIconGhostVisibility;
     public override void Initialize()
     {
         base.Initialize();
 
-        SubscribeLocalEvent<RevolutionaryComponent, GetStatusIconsEvent>(GetRevIcon);
-        SubscribeLocalEvent<HeadRevolutionaryComponent, GetStatusIconsEvent>(GetHeadRevIcon);
+        _cfg.OnValueChanged(CCVars.RevIconsVisibleToGhosts, value => _revIconGhostVisibility = value, true);
+        SubscribeLocalEvent<RevolutionaryComponent, CanDisplayStatusIconsEvent>(OnCanShowRevIcon);
```

Comment:
This event will leak, you need to unsubscribe on the client.

Reply:
This is the client. This event is needed to actually draw the icons.